core/src/main/scala/org/apache/spark/sql/TiContext.scala

```diff
@@ -49,11 +52,15 @@ class TiContext(val sparkSession: SparkSession) extends Serializable with Loggin
     } else Option.empty)
   final val tiSession: TiSession = TiSession.getInstance(tiConf)
   lazy val sqlContext: SQLContext = sparkSession.sqlContext
+  val defaultTiSparkGCSafePointTTL: Int = 5 * 60
+  val serviceSavePoint: ServiceSavePoint =
+    ServiceSavePoint("tispark_" + UUID.randomUUID, defaultTiSparkGCSafePointTTL, tiSession)
 
   sparkSession.sparkContext.addSparkListener(new SparkListener() {
     override def onApplicationEnd(applicationEnd: SparkListenerApplicationEnd): Unit = {
       if (tiSession != null) {
         try {
+          serviceSavePoint.stopRegisterSavePoint()
```

Comment:
if the application end, why not remove the minstartts instantly?

Reply:
good suggestion. But we can not remove it for there is no API. we can set a Long.MAXVALUE